plugins/src/main/java/org/tron/plugins/utils/MerkleTree.java

```diff
@@ -0,0 +1,94 @@
+package org.tron.plugins.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import lombok.Getter;
+
+@Getter
+public class MerkleTree {
+
+  private static volatile MerkleTree instance;
+  private List<Sha256Hash> hashList;
+  private List<Leaf> leaves;
+  private Leaf root;
+
+  public static MerkleTree getInstance() {
```

Comment:
here it uses Double-Checked Locking to ensure that only one instance is created in a multi-threaded environment. However, this code suffers from concurrency issues. Double-Checked Locking is not always reliable in Java because it depends on the memory model and compiler optimizations.
A safer implementation of the singleton pattern uses static inner classes in a way that is thread-safe and efficient。
`   private static class MerkleTreeHolder {
        private static final MerkleTree instance = new MerkleTree();
    }

    public static MerkleTree getInstance() {
        return MerkleTreeHolder.instance;
    }`

Reply:
This class has member variables and can not be used for concurrency,  need to modify the usage scenarios while labeling them as not being able to be concurrent.